### PR TITLE
Use `max` (instead of `sum`) for calculating `warnsize`

### LINKED ISSUE #9146: wrong calculation of `warnsize` in slicing?
**What happened**:
```
PerformanceWarning: Slicing is producing a large chunk.
```

**What you expected to happen**:
No performance warning. Resulting chunks are each below 10MiB.

**Minimal Complete Verifiable Example**:

```python
import numpy as np
import dask.array as da
a = da.zeros((100000, 10000000), chunks=(1, 10000000))
a[:,np.arange(1000000)]
```

**Anything else we need to know?**:

The calculation of `warnsize` at [`take`](https://github.com/dask/dask/blob/cccb9d8d8e33a891396b1275c2448c352ef40c27/dask/array/slicing.py#L642) might be wrong. My guess would be, that it should be `max` instead of `sum` like:

```python
np.prod([max(x) for x in other_chunks])
```

That way, the largest resulting chunk would be calculated (instead of chunksize * shape of other dimensions).

**Environment**:

- Dask version: 2022.05.2
- Python version: 3.9.9
- Operating System: Linux
- Install method (conda, pip, source): pip

### PR DESCRIPTION
**Pull request by pavithraes**
- [x] Closes #9146
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/array/slicing.py [hunk 1]
```diff
@@ -639,7 +639,7 @@ def take(outname, inname, chunks, index, itemsize, axis=0):
     # configured chunk size.
     nbytes = utils.parse_bytes(config.get("array.chunk-size"))
     other_chunks = [chunks[i] for i in range(len(chunks)) if i != axis]
-    other_numel = math.prod(sum(x) for x in other_chunks)
+    other_numel = math.prod([max(x) for x in other_chunks])
```
Comment (jrbourbeau) [inline dask/array/slicing.py:642]: Just curious, is there a reason we need to fully materialize a this list before passing to `math.prod`? 

```suggestion
    other_numel = math.prod(max(x) for x in other_chunks)
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @pavithraes! It looks like we're currently warning when we shouldn't. Could we add a small regression test to make sure we don't emit a warning in that case with the changes here?

File: dask/array/slicing.py (hunk 1 above)
Comment (pavithraes) [reply dask/array/slicing.py]: Oh, thanks for catching! I copied over the first commit changes after merging `main`, and didn't realize [we had updated this too](https://github.com/dask/dask/pull/9232/files#diff-cc8bb1c7ea02a9582c31ac1580f3ab0674c35e0b6af060513ba9932ceb07a600L642).

**Comment by pavithraes** [comment]
> It looks like we're currently warning when we shouldn't. Could we add a small regression test to make sure we don't emit a warning in that case with the changes here?

Yes! Added. :)

**Review by jrbourbeau — APPROVED** [review]
Thanks @pavithraes! This is in

**Pull request merged**